backend/de.metas.adempiere.adempiere/base/src/main/java/de/metas/i18n/ITranslatableString.java

```diff
@@ -44,4 +45,6 @@ default boolean isTranslatedTo(final String adLanguage)
 	{
 		return getAD_Languages().contains(adLanguage);
 	}
+
+	default @Nullable String getErrorCode() {return null;}
```

Comment:
i don't feel OK to have the getErrorCode() here. ITranslatableString is supposed to be quite slim and simple.

What's the precise case that leads to this?

Why not simply having errorCode in AdempiereException is not enough? why it had to be spread?

Reply:
Main usage for this would be so that `ADElementOrADMessageTranslatableString`/`ADMessageTranslatableString` can offer an error code when used as part of `new AdempiereException(ITranslateableString)`. Without this, the error code would be lost if building a translateable string from an AdMessageKey to pass into adempiere exception.

Reply:
removed, and replaced where possible usages for `new AdempiereException(ITranslateableString)` with `new AdempiereException(AdMessageKey)`